h2o-core/src/main/java/water/rapids/Rapids.java

```diff
@@ -67,6 +69,9 @@ public static Val exec(String rapids) {
     } catch (Throwable ex) {
       throw session.endQuietly(ex);
     }
+    finally {
```

Comment:
I would personally not record end of rapids execution, since we can to track while it is in progress. At least start is well defined. Otherwise you can have racy behavior where cluster might or might not be shut down depending on whether some long running rapids expression finished before time-out or not.